src/browser/screens/Roles.react.js

```diff
@@ -20,17 +22,23 @@ export default class ThirdScreen extends Component {
 
   getDefaultState() {
     return {
-      role: null
+      role: null,
+      select: false,
+      current: ''
     };
   }
 
   proceed() {
-    if (!this.state.role)
-      return;
-
-    const {actions: {saveRole, nextScreen}} = this.props;
-    saveRole(this.state.role);
-    nextScreen();
+    if (!this.state.role) {
+      this.setState({select: "select"});
+      setTimeout(() => { this.setState({select: false}); }, 5000);
+    } else {
+      const {actions: {saveRole, nextScreen}} = this.props;
+      saveRole(this.state.role);
+      this.setState({current: this.state.role.id});
+      this.setState({select: false});
```

Comment:
`{current: this.state.role.id, select: false}` Nie ma potrzeby wywolywac dwa razy pod rzad `setState`